src/unique_impl.rs

```diff
@@ -133,14 +133,15 @@ where
     I::Item: Eq + Hash + Clone,
 {
     fn next_back(&mut self) -> Option<Self::Item> {
-        while let Some(v) = self.iter.iter.next_back() {
-            if let Entry::Vacant(entry) = self.iter.used.entry(v) {
+        let UniqueBy { iter, used, .. } = &mut self.iter;
+        iter.rev().find_map(|v| {
```

Comment:
Off-topic: Is it strange that there's `rfind`, but no `rfind_map`?

Reply:
It's not off-topic to me. I was wondering the same question while writing this.
Maybe there is no large performance gain to expect compare to `.rev().method()`. I searched a bit but did not find anything.
More generally, `DoubleEndedIterator` does not have a lot of methods.